File: client/java/src/main/java/com/be/mesh/client/prsim/Evaluator.java
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */
package com.be.mesh.client.prsim;

import com.be.mesh.client.annotate.MPI;
import com.be.mesh.client.annotate.SPI;
import com.be.mesh.client.struct.Page;
import com.be.mesh.client.struct.Paging;
import com.be.mesh.client.struct.Script;

import java.util.List;
import java.util.Map;

/**
 * @author coyzeng@gmail.com
 */
@SPI("mesh")
public interface Evaluator {

    /**
     * Compile the named rule.
     */
    @MPI("mesh.eval.compile")
    String compile(Script script);

    /**
     * Exec the script with code.
     */
    @MPI("mesh.eval.exec")
    String exec(String code, Map<String, String> args, String dft);

    /**
     * Dump the scripts.
     */
    @MPI("mesh.eval.dump")
    List<Script> dump(Map<String, String> feature);

    /**
     * Index the scripts.
     */
    @MPI("mesh.eval.index")
    Page<Script> index(Paging index);
}

File: plugin/prsim/evaluator.go
/*
 * Copyright (c) 2000, 2023, trustbe and/or its affiliates. All rights reserved.
 * TRUSTBE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 *
 *
 */

package prsim

import (
	"context"
	"github.com/be-io/mesh/client/golang/prsim"
	"github.com/be-io/mesh/client/golang/types"
)

var _ prsim.Evaluator = new(PSRIEvaluator)

// PSRIEvaluator
// @MPS(macro.MeshMPS)
// @SPI(macro.MeshSPI)
type PSRIEvaluator struct {
}

func (that *PSRIEvaluator) Compile(ctx context.Context, script *types.Script) (string, error) {
	return aware.Evaluator.Compile(ctx, script)
}

func (that *PSRIEvaluator) Exec(ctx context.Context, code string, args map[string]string, dft string) (string, error) {
	return aware.Evaluator.Exec(ctx, code, args, dft)
}

func (that *PSRIEvaluator) Dump(ctx context.Context, feature map[string]string) ([]*types.Script, error) {
	return aware.Evaluator.Dump(ctx, feature)
}

func (that *PSRIEvaluator) Index(ctx context.Context, index *types.Paging) (*types.Page, error) {
	return aware.Evaluator.Index(ctx, index)
}
